homeassistant/components/broadlink/time.py

```diff
@@ -0,0 +1,63 @@
+"""Support for Broadlink device time."""
+
+from __future__ import annotations
+
+from datetime import time
+from typing import Any
+
+from homeassistant.components.time import TimeEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util import dt as dt_util
+
+from . import BroadlinkDevice
+from .const import DOMAIN
+from .entity import BroadlinkEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Broadlink time."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+    async_add_entities([BroadlinkTime(device)])
+
+
+class BroadlinkTime(BroadlinkEntity, TimeEntity):
```

Comment:
Let's name this entity BroadlinkHysenTime, as it is specific to the Hysen class, and we may want to add other types in the future.

Reply:
Do you think other device types would have a fundamentally different time platform? The data used is retrieved via the broadlink api `get_full_status`

Reply:
Yes, although we currently only have this device with day of the week and time in the API today, this structure could vary depending on the type of device in the future. Some devices might include day, month, and year, while others might include timezone information.

Reply:
Do we know of a device which is available today or will be available in the future? Or is this an assumption?

Reply:
It's always an assumption when it comes to Broadlink. When we don't know what is going to happen, we better keep it flexible. But I am totally okay if you want to leave it as is and change it later if we need to.